spec/unit/ace/transport_app_spec.rb

```diff
@@ -278,4 +279,172 @@ def app
       end
     end
   end
+
+  ##################
+  # init_puppet_target function
+  ##################
+  describe 'init_puppet_target' do
+    describe 'success with transport style connection info' do
+      device_raw = '{
+            "target": {
+                "remote-transport":"panos",
+                "host":"fw.example.net",
+                "user":"foo",
+                "password":"wibble"
+            },
+            "compiler": {
+                "certname":"fw.example.net",
+                "environment":"development",
+                "transaction_uuid":"<uuid string>",
+                "job_id":"<id string>"
+            }
+        }'
+      device_json = JSON.parse(device_raw)
+      device_return_struct = Struct.new(:current)
+      it 'returns correct device' do
+        allow(Puppet::Util::NetworkDevice).to receive(:init) do |params|
+          expect(params[:provider]).to eq(device_json['target']['remote-transport'])
+          expect(params[:url]).to eq(Hash[device_json['target'].map { |(k, v)| [k.to_sym, v] }])
+          expect(params[:name]).to eq(device_json['compiler']['certname'])
+          expect(params[:options]).to eql({})
+          device_return_struct.new('panos_current')
+        end
+
+        expect(described_class.init_puppet_target(device_json['compiler']['certname'],
+                                                  device_json['target']['remote-transport'],
+                                                  device_json['target'])).to match(/(panos_current)/)
+      end
+    end
+
+    describe 'success with legacy style uri' do
+      device_raw = '{
+          "target":{
+            "remote-transport":"f5",
+            "uri":"https://foo:wibble@f5.example.net/"
+          },
+          "compiler":{
+            "certname":"f5.example.net",
+            "environment":"development",
+            "transaction_uuid":"<uuid string>",
+            "job_id":"<id string>"
+          }
+        }'
+      device_json = JSON.parse(device_raw)
+      device_return_struct = Struct.new(:current)
+      it 'returns correct device' do
+        allow(Puppet::Util::NetworkDevice).to receive(:init) do |params|
+          expect(params[:provider]).to eq(device_json['target']['remote-transport'])
+          expect(params[:url]).to eq(device_json['target']['uri'])
+          expect(params[:name]).to eq(device_json['compiler']['certname'])
+          expect(params[:options]).to eql({})
+          device_return_struct.new('f5_current')
```

Comment:
Not sure how I feel about this. While you are mocking the response, the value returned by init is a device object and not a struct. If someone is looking at the tests they may get the wrong idea. So perhaps add a comment about this to say that this will (in practice) return a device object, while at line 310 it will return a transport object?